src/common-ssh/ssh.c

```diff
@@ -518,6 +520,40 @@ guac_common_ssh_session* guac_common_ssh_create_session(guac_client* client,
         return NULL;
     }
 
+    /* Get fingerprint of host we're connecting to */
+    size_t fp_len;
+    int fp_type;
+    const char *fingerprint = libssh2_session_hostkey(session, &fp_len, &fp_type);
+
+    /* Failure to generate a fingerprint means we should abort */
+    if (!fingerprint) {
+        guac_client_abort(client, GUAC_PROTOCOL_STATUS_SERVER_ERROR,
+            "Failed to get fingerprint for host %s", hostname);
+        return NULL;
```

Comment:
This will leak `common_session` and `fd`, allocated earlier in this function. For example, see:

https://github.com/apache/guacamole-server/blob/c120aa0274ee2d11465c80e1d3052c725db44a17/src/common-ssh/ssh.c#L512-L519

Reply:
Leak plugged.